base/math.jl

```diff
@@ -27,6 +27,16 @@ using Core.Intrinsics: sqrt_llvm
 
 const IEEEFloat = Union{Float16, Float32, Float64}
 
+@noinline function throw_complex_domainerror(f, x)
+    throw(DomainError(x, string("$f will only return a complex result if called with a ",
+                                "complex argument. Try $f(complex(x)).")))
+end
+@noinline function throw_exp_domainerror(x)
+    throw(DomainError(x, string("Exponentiation yielding a complex result requires a ",
+                                "complex argument.\nReplace x^y with (x+0im)^y, ",
+                                "Complex(x)^y, or similar.")))
```

Comment:
The first method recommends calling `complex`, whereas the second recommends calling `Complex`; should these methods consistently recommend one or the other?